sample/src/main/java/com/github/barteksc/sample/PDFViewActivity.java

```diff
@@ -74,84 +89,97 @@ public class PDFViewActivity extends AppCompatActivity implements OnPageChangeLi
+    }
+
     @AfterViews
     void afterViews() {
         pdfView.setBackgroundColor(Color.LTGRAY);
-        if (uri != null) {
-            displayFromUri(uri);
-        } else {
-            displayFromAsset(SAMPLE_FILE);
-        }
         setTitle(pdfFileName);
     }
 
-    private void displayFromAsset(String assetFileName) {
-        pdfFileName = assetFileName;
-
-        pdfView.fromAsset(SAMPLE_FILE)
-                .defaultPage(pageNumber)
-                .onPageChange(this)
-                .enableAnnotationRendering(true)
-                .onLoad(this)
-                .scrollHandle(new DefaultScrollHandle(this))
-                .spacing(10) // in dp
-                .onPageError(this)
-                .onTap(this)
-                .onLongPress(this)
-                .load();
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode == RESULT_OK && requestCode == PublicValue.KEY_REQUEST_FILE_PICKER) {
+            if (data != null && data.getData() != null) {
+                this.currUri = data.getData();
+                displayFileFromUri();
+            } else {
+                Log.e(TAG, "onActivityResult, requestCode:" + requestCode + "resultCode:" + resultCode);
+            }
+        }
     }
 
-    private void displayFromUri(Uri uri) {
-        pdfFileName = getFileName(uri);
+    private void displayFileFromUri() {
+
+        if (currUri == null) {
+            com.github.barteksc.sample.DebugUtilKt.toast(this, "currUri is null");
+            return;
+        }
+
+        // TODO: 1/17/21  DON NOT FORGET TO USE YOUR FILE HANDLING SCENARIO FOR NEW ANDROID APIs
+
+        // this is OK
+        // /storage/emulated/0/Download/PDF_ENGLISH.pdf
+        // this.currFilePath = UriUtils.getPathFromUri(MainActivity.this, currUri);
+
+        // this is not OK
+        // /data/user/0/ir.vasl.magicalpdfeditor/files/PDF_ENGLISH.pdf
+        // this.currFilePath = PublicFunction.getFilePathForN(MainActivity.this, currUri);
+
+        // this is working
+        // /storage/emulated/0/Download/PDF_ENGLISH.pdf
 
-        pdfView.fromUri(uri)
-                .defaultPage(pageNumber)
+        pdfFileName = getFileName(currUri);
+
+        this.configurator = pdfView.fromUri(currUri)
+                .defaultPage(PublicValue.DEFAULT_PAGE_NUMBER)
                 .onPageChange(this)
                 .enableAnnotationRendering(true)
+                .enableAnnotationHandling(true)
                 .onLoad(this)
+                .enableSwipe(true)
                 .scrollHandle(new DefaultScrollHandle(this))
                 .spacing(10) // in dp
                 .onPageError(this)
                 .onTap(this)
                 .onLongPress(this)
-                .load();
-    }
+                .onError(this)
+                .linkHandler(this);
```

Comment:
`linkHandler` is `Link` annotation handler, `linkHandler` will do nothing since we won't use Link annotations

We still have the problem (coming from PDFs nature) that when tapping on the `PdfView` we can only check if there is an annotation (only `Link` in our case) in this region but we aren't clicking on the annotation itself. 
 
I wonder how we will detect the tapped annotation, how can we use measureLib again. If we pass the coordinates of the touched point can we select the shape programatically in measureLib and trigger the dialog, finish `MeasureActivity` but leave the dialog open, will something like this work?

Reply:
> We still have the problem (coming from PDFs nature) that when tapping on the PdfView we can only check if there is an annotation (only Link in our case) in this region but we aren't clicking on the annotation itself.

We can access all annotations and check if the touch point belongs to some annotation I believe but thats a thing we will take a look once we need this

> . If we pass the coordinates of the touched point can we select the shape programatically in measureLib and trigger the dialog, finish MeasureActivity but leave the dialog open, will something like this work?

I dont think will look smoothly

Reply:
I got your idea.

If I change `LinkHandler` and  `LinkTapEvent` to not use link annotation,  `LinkTapEvent`  basically becomes a wrapper over the screen and pdf coordinates of the touched point, and the `LinkHandler` uses the `LinkTapEvent`. 

> If we want to open the linked entities dialog when clicking on an annotation we gonna need a way to detect the click 

Do we want to do that on a tap or on a long click?  We can keep the link handler if we want to do that on tap or remove it if we want to do it on long click. `LinkHandler` was set to work for a tap . We can remove one of them 

 > access all annotations and check if the touch point belongs to some annotation
 
 this sounds good

Reply:
> Do we want to do that on a tap or on a long click?

That's up for discussion but I would say on a tap.

But that's not something we need to worry about here - we will tackle it when that time arrives

Reply:
>  I would say on a tap

Keeping the link handler then to have on tap. Changed `LinkHandler` and `DefaultLinkHandler` to work without link annotation. Using the `DefaultLinkHandler` we don't need to override anything in the activity.